src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -1005,6 +1018,11 @@ export class ShadowGenerator implements IShadowGenerator {
     }
 
     protected _bindCustomEffectForRenderSubMeshForShadowMap(subMesh: SubMesh, effect: Effect): void {
+        effect.setMatrix("viewProjection", this.getTransformMatrix());

```

Comment:
Why do we need this change?

Reply:
I moved those binding in this function because those are overwritten by the CSM class because CSM provides its own implementation of `_bindCustomEffectForRenderSubMeshForShadowMap`. It avoids having those uniforms set by the base class then being immediately overwritten by CSM (double work for nothing).

Reply:
Ok makes sense!